src/index.ts

```diff
@@ -1,10 +1,17 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import { createIoCContainer } from './ioc';
+import type { Users } from './services/users';
+import type { User } from './types';
+const ioc = createIoCContainer();
 
-import type { User, ApiConfig } from './types';
+window.onload = (event: Event) => {
+  const logger = ioc.resolve('logger');
+  logger.info('Page is loaded.');
+
+  const usersService = ioc.resolve('users') as Users;
```

Comment:
Can you add generic type to your ioc container like this `IoCContainer<IoCResources>` instead of using `as Users;`.
you can find an example here: https://www.npmjs.com/package/ioc-lite#basic-usage